ayaneo-platform.c

```diff
@@ -927,11 +966,391 @@ struct led_classdev_mc ayaneo_led_mc = {
+                                        ayaneo_bypass_charge_legacy_open();
+                                        break;
+                                case air_plus:
+                                case slide:
+                                        ayaneo_bypass_charge_open();
+                                        break;
+                                default:
+                                        break;
+                            }
+                        } else if(capacity.intval < ps_priv.current_start_threshold) {
+                            switch (model) {
+                                case air:
+                                case air_1s:
+                                case air_1s_limited:
+                                case air_pro:
+                                case air_plus_mendo:
+                                case geek:
+                                case geek_1s:
+                                case ayaneo_2:
+                                case ayaneo_2s:
+                                case kun:
+                                        ayaneo_bypass_charge_legacy_close();
+                                        break;
+                                case air_plus:
+                                case slide:
+                                        ayaneo_bypass_charge_close();
+                                        break;
+                                default:
+                                        break;
+                            }
+                        }
+                    }
+                } else if (last_charge_type != ps_priv.charge_types){
+                    switch (model) {
+                        case air:
+                        case air_1s:
+                        case air_1s_limited:
+                        case air_pro:
+                        case air_plus_mendo:
+                        case geek:
+                        case geek_1s:
+                        case ayaneo_2:
+                        case ayaneo_2s:
+                        case kun:
+                                ayaneo_bypass_charge_legacy_close();
+                                break;
+                        case air_plus:
+                        case slide:
+                                ayaneo_bypass_charge_close();
+                                break;
+                        default:
+                                break;
+                    }
+                }
+                last_charge_type = ps_priv.charge_types;
+                msleep(AYANEO_BYPASS_WRITER_DELAY_MS);
+            }
+
+            pr_info("Bypass-Writer thread stopped.\n");
+        }
+        return 0;
+}
+
+static enum power_supply_property ayaneo_ps_props[] = {
+    POWER_SUPPLY_PROP_CHARGE_TYPE,
+    POWER_SUPPLY_PROP_CHARGE_CONTROL_START_THRESHOLD,
+    POWER_SUPPLY_PROP_CHARGE_CONTROL_END_THRESHOLD,
+};
+
+struct power_supply_desc ayaneo_ps = {
+    .name       = "ayaneo:bypass_charge",
```

Comment:
Is the name reasonable?

Reply:
See my review comment on attaching to the battery